repos/system_upgrade/el7toel8/actors/restrictedpcisscanner/libraries/restrictedpcisscanner.py

```diff
@@ -0,0 +1,105 @@
+import json
+import os
+
+import urllib3
+
+from leapp.libraries.common.config import get_env
+from leapp.libraries.stdlib import api
+from leapp.models import RestrictedPCIDevices
+
+REQUEST_TIMEOUT = 0.5
+CUR_DIR = os.path.dirname(os.path.abspath(__file__))
+UNSUPPORTED_DRIVER_NAMES_FILE = os.path.join(
+    CUR_DIR, "..", "data", "unsupported_driver_names.json"
+)
+UNSUPPORTED_PCI_IDS_FILE = os.path.join(
+    CUR_DIR, "..", "data", "unsupported_pci_ids.json"
+)
+
+
+def get_restricted_pcis(http, update_local_data=False):
+    """Get data about unsupported driver names and unsupported pci_ids.
+
+    The function makes API call to the web service, which provides
+    these data.
+
+    :param urllib3.PoolManager http: pool manager instance
+    :param bool update_local_data: if True writes the received data to
+        the corresponding json data files (inside ../data dir)
+
+    :returns: Tuple[dict, dict]
+    """
+    # TODO this constant should be above the scope of this func.
+    #   However due to the way how actor tested with CurrentActorMocked
+    #   we can't do this.
+    API_URL = (
+        "http://"
+        + get_env(
+            "LEAPP_DEVEL_UNSUPPORTED_PCIS_HOST", default="10.0.79.153:8000"
+        )
+        + "/api/v1/"
+    )
+    unsupported_driver_names = http.request(
```

Comment:
Nit: can be refactored into a function and reused

Reply:
I tried and the solution eventually has more overhead then advantages (IMHO).
- additional function
- additional conditions 

Is it a principal requirement?